src/IceRpc.Retry/RetryInterceptor.cs

```diff
@@ -43,8 +43,9 @@ namespace IceRpc.Retry;
 public class RetryInterceptor : IInvoker
 {
     private readonly ILogger _logger;
+    private readonly int _maxAttempts;
```

Comment:
I think the preferred pattern is / should be to _not_ keep a reference to options (which the caller could in theory change) but instead copy options into individual fields. That's what we do already for Ice and IceRpcProtocolConnection.